library/Vanilla/Formatting/DateTimeFormatter.php

```diff
@@ -0,0 +1,406 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting;
+
+use Garden\StaticCacheTranslationTrait;
+
+/**
+ * Formatting methods related to dates & times.
+ */
+class DateTimeFormatter {
+
+    use StaticCacheTranslationTrait;
+
+    const FORCE_FULL_FORMAT = 'force-full-datetime-format';
+
+    /** @var DateConfig */
+    private $dateConfig;
+
+    /**
+     * @param DateConfig $dateConfig
+     */
+    public function __construct(DateConfig $dateConfig) {
+        $this->dateConfig = $dateConfig;
+    }
+
+    const NULL_TIMESTAMP_DEFALT_VALUE = '-';
```

Comment:
Could we keep `const`s declaration on the top before class methods starts and before properties  please?

Reply:
Done